### PR TITLE
[2.7] bpo-8765: Deprecate writing unicode to binary streams in Py3k mode.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
https://bugs.python.org/issue8765

### CODE REVIEW

**Comment by benjaminp** [comment]
I wonder if t might be cheaper to check `Py_TYPE(PyTuple_GET_ITEM(args, 0))` after doing the initial parsing?

**Comment by serhiy-storchaka** [comment]
Good point!

File: Lib/test/test_fileio.py [hunk 1]
```diff
@@ -101,6 +101,10 @@ def test_none_args(self):
         self.assertEqual(self.f.readline(None), b"hi\n")
         self.assertEqual(self.f.readlines(None), [b"bye\n", b"abc"])
 
+    def testWriteUnicode(self):
+        with check_py3k_warnings():
+            self.f.write('')
```
Comment (benjaminp) [inline Lib/test/test_fileio.py]: Even though this file is using `unicode_literals`, it wouldn't hurt to write `u""` here.

**Review by benjaminp — APPROVED** [review]

File: Lib/test/test_fileio.py (hunk 1 above)
Comment (serhiy-storchaka) [inline Lib/test/test_fileio.py]:
```suggestion
            self.f.write(u'')
```

**Pull request merged**